src/mlpack/methods/ann/layer/padding_impl.hpp

```diff
@@ -42,10 +46,32 @@ void Padding<InputDataType, OutputDataType>::Forward(
 {
   nRows = input.n_rows;
   nCols = input.n_cols;
-  output = arma::zeros(nRows + padWLeft + padWRight,
-      nCols + padHTop + padHBottom);
-  output.submat(padWLeft, padHTop, padWLeft + nRows - 1,
-      padHTop + nCols - 1) = input;
+  
+  if (inputWidth == 0 || inputHeight == 0)
+  {
+    output = arma::zeros(nRows + padWLeft + padWRight,
+        nCols + padHTop + padHBottom);
+    output.submat(padWLeft, padHTop, padWLeft + nRows - 1,
+        padHTop + nCols - 1) = input;
+  }
+  else
+  {
+    inSize = input.n_elem / (inputWidth * inputHeight * nCols);
+    inputTemp = arma::Cube<eT>(const_cast<arma::Mat<eT>&>(input).memptr(),
+        inputWidth, inputHeight, inSize * nCols, false, false);
+    outputTemp = arma::zeros<arma::Cube<eT> >(inputWidth + padWLeft + padWRight,
+        inputHeight + padHTop + padHBottom, inSize * nCols);
+    for (size_t i = 0; i < inputTemp.n_slices; ++i)
+    {   
+      outputTemp.slice(i).submat(padWLeft, padHTop, padWLeft + inputWidth - 1,
+          padHTop + inputHeight - 1) = inputTemp.slice(i);
+    }
```

Comment:
Looks good, Could you share a print output for a test case [Output of each slice]. Just to verify that values are being set in the cells.

Reply:
Sure, give me a minute.

Reply:
Code used, i tried to keep it simple with the dimensions so it's easy to check
```c++
#include <mlpack/core.hpp>
#include <mlpack/methods/ann/layer/layer.hpp>
#include <mlpack/methods/ann/ffn.hpp>
#include <mlpack/methods/ann/layer/layer_types.hpp>

using namespace mlpack;
using namespace arma;

int main()
{
  arma::mat input, output;
  ann::Padding<> module(1, 1, 1, 1, 3, 3);
  input = arma::randu(3 * 3 * 1, 1);

  module.Forward(input, output);

  std::cout<<"input Rows: "<<input.n_rows<<std::endl;
  std::cout<<"input Cols: "<<input.n_cols<<std::endl;
  input.print();

  std::cout<<"output Rows: "<<output.n_rows<<std::endl;
  std::cout<<"output Cols: "<<output.n_cols<<std::endl;
  output.print();

  return 0;
}
```
Not the best visualization in the world but does the job, 
```
input Rows: 9
input Cols: 1
   0.7868
   0.2505
   0.7107
   0.9467
   0.0193
   0.4049
   0.2513
   0.0227
   0.5206
output Rows: 25
output Cols: 1
        0
        0
        0
        0
        0
        0
   0.7868
   0.2505
   0.7107
        0
        0
   0.9467
   0.0193
   0.4049
        0
        0
   0.2513
   0.0227
   0.5206
        0
        0
        0
        0
        0
        0
```